testscommon/cacherMock.go

```diff
@@ -0,0 +1,136 @@
+package testscommon
+
+import (
+	"sync"
+
+	"github.com/multiversx/mx-chain-storage-go/types"
+)
+
+type cacherMock struct {
+	mut  sync.RWMutex
+	data map[string]interface{}
+}
+
+// NewCacherMock -
+func NewCacherMock() *cacherMock {
+	return &cacherMock{
+		data: make(map[string]interface{}),
+	}
+}
+
+// Clear -
+func (mock *cacherMock) Clear() {
+	mock.mut.Lock()
+	defer mock.mut.Unlock()
+
+	mock.data = make(map[string]interface{})
+}
+
+// Put -
+func (mock *cacherMock) Put(key []byte, value interface{}, _ int) (evicted bool) {
+	mock.mut.Lock()
+	defer mock.mut.Unlock()
+
+	mock.data[string(key)] = value
+
+	return false
+}
+
+// Get -
+func (mock *cacherMock) Get(key []byte) (value interface{}, ok bool) {
+	mock.mut.RLock()
+	defer mock.mut.RUnlock()
+
+	val, found := mock.data[string(key)]
+
+	return val, found
+}
+
+// Has -
+func (mock *cacherMock) Has(key []byte) bool {
+	mock.mut.RLock()
+	defer mock.mut.RUnlock()
+
+	_, found := mock.data[string(key)]
+
+	return found
+}
+
+// Peek -
+func (mock *cacherMock) Peek(key []byte) (value interface{}, ok bool) {
+	return mock.Get(key)
+}
+
+// HasOrAdd -
+func (mock *cacherMock) HasOrAdd(key []byte, value interface{}, _ int) (has, added bool) {
+	mock.mut.Lock()
+	defer mock.mut.Unlock()
+
+	_, found := mock.data[string(key)]
+
```

Comment:
Missing here: if found { return found, !found}  ?

Reply:
right, changed also the other return as it was badly written